expanding(min_periods=N).count(), for any N, returns same result as expanding(min_periods=1).count()

```python
idx=pd.date_range('1/1/2019', periods=5)
s = pd.Series([2,3, np.nan, 10,20], index=idx)
s.expanding(min_periods=1).count()
s.expanding(min_periods=100).count()
s.expanding(min_periods=10000).count()

```
#### Problem description
They should not be returning the same results because the Series has only 5 records. 

**Note**: We receive a lot of issues on our GitHub tracker, so it is very possible that your issue has been posted before. Please check first before submitting so that we do not have to handle and close duplicates!

**Note**: Many problems can be resolved by simply upgrading `pandas` to the latest version. Before submitting, please check if that solution works for you. If possible, you may want to check if `master` addresses this issue, but that is not necessary.

For documentation-related issues, you can check the latest versions of the docs on `master` here:

https://pandas-docs.github.io/pandas-docs-travis/

If the issue has not been resolved there, go ahead and file it in the issue tracker.

#### Expected Output

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]

</details>

Changed to the following lines, it willl break pandas\tests\test_window.py. The test is expecting to have none NaN values as if min_window is always set to 0

            result = self._constructor(result, window=window,
                                       min_periods=self.min_periods
```ipython
====================================================== FAILURES =======================================================
_________________________________________ TestApi.test_count_nonnumeric_types _________________________________________

self = <pandas.tests.test_window.TestApi object at 0x0000020B27671D30>

    def test_count_nonnumeric_types(self):
        # GH12541
        cols = ['int', 'float', 'string', 'datetime', 'timedelta', 'periods',
                'fl_inf', 'fl_nan', 'str_nan', 'dt_nat', 'periods_nat']

        df = DataFrame(
            {'int': [1, 2, 3],
             'float': [4., 5., 6.],
             'string': list('abc'),
             'datetime': pd.date_range('20170101', periods=3),
             'timedelta': pd.timedelta_range('1 s', periods=3, freq='s'),
             'periods': [pd.Period('2012-01'), pd.Period('2012-02'),
                         pd.Period('2012-03')],
             'fl_inf': [1., 2., np.Inf],
             'fl_nan': [1., 2., np.NaN],
             'str_nan': ['aa', 'bb', np.NaN],
             'dt_nat': [Timestamp('20170101'), Timestamp('20170203'),
                        Timestamp(None)],
             'periods_nat': [pd.Period('2012-01'), pd.Period('2012-02'),
                             pd.Period(None)]},
            columns=cols)

        expected = DataFrame(
            {'int': [1., 2., 2.],
             'float': [1., 2., 2.],
             'string': [1., 2., 2.],
             'datetime': [1., 2., 2.],
             'timedelta': [1., 2., 2.],
             'periods': [1., 2., 2.],
             'fl_inf': [1., 2., 2.],
             'fl_nan': [1., 2., 1.],
             'str_nan': [1., 2., 1.],
             'dt_nat': [1., 2., 1.],
             'periods_nat': [1., 2., 1.]},
            columns=cols)

        result = df.rolling(window=2).count()
>       tm.assert_frame_equal(result, expected)

pandas\tests\test_window.py:280:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pandas\_libs\testing.pyx:65: in pandas._libs.testing.assert_almost_equal
    cpdef assert_almost_equal(a, b,
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   raise_assert_detail(obj, msg, lobj, robj)
E   AssertionError: DataFrame.iloc[:, 0] are different
E
E   DataFrame.iloc[:, 0] values are different (33.33333 %)
E   [left]:  [nan, 2.0, 2.0]
E   [right]: [1.0, 2.0, 2.0]

pandas\_libs\testing.pyx:178: AssertionError
____________________________________________ TestDtype_object.test_dtypes _____________________________________________

self = <pandas.tests.test_window.TestDtype_object object at 0x0000020B27B97320>

    def test_dtypes(self):
        self._create_data()
        for f_name, d_name in product(self.funcs.keys(), self.data.keys()):

            f = self.funcs[f_name]
            d = self.data[d_name]
            exp = self.expects[d_name][f_name]
>           self.check_dtypes(f, f_name, d, d_name, exp)

pandas\tests\test_window.py:938:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pandas\tests\test_window.py:943: in check_dtypes
    tm.assert_almost_equal(result, exp)
pandas\util\testing.py:296: in assert_almost_equal
    **kwargs)
pandas\_libs\testing.pyx:65: in pandas._libs.testing.assert_almost_equal
    cpdef assert_almost_equal(a, b,
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   raise_assert_detail(obj, msg, lobj, robj)
E   AssertionError: Series are different
E
E   Series values are different (20.0 %)
E   [left]:  [nan, 2.0, 2.0, 2.0, 2.0]
E   [right]: [1.0, 2.0, 2.0, 2.0, 2.0]

pandas\_libs\testing.pyx:178: AssertionError
_____________________________________________ TestDtype_int8.test_dtypes ______________________________________________

self = <pandas.tests.test_window.TestDtype_int8 object at 0x0000020B27780438>

    def test_dtypes(self):
        self._create_data()
        for f_name, d_name in product(self.funcs.keys(), self.data.keys()):

            f = self.funcs[f_name]
            d = self.data[d_name]
            exp = self.expects[d_name][f_name]
>           self.check_dtypes(f, f_name, d, d_name, exp)

pandas\tests\test_window.py:938:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pandas\tests\test_window.py:943: in check_dtypes
    tm.assert_almost_equal(result, exp)
pandas\util\testing.py:296: in assert_almost_equal
    **kwargs)
pandas\_libs\testing.pyx:65: in pandas._libs.testing.assert_almost_equal
    cpdef assert_almost_equal(a, b,
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   raise_assert_detail(obj, msg, lobj, robj)
E   AssertionError: Series are different
E
E   Series values are different (20.0 %)
E   [left]:  [nan, 2.0, 2.0, 2.0, 2.0]
E   [right]: [1.0, 2.0, 2.0, 2.0, 2.0]

pandas\_libs\testing.pyx:178: AssertionError
_____________________________________________ TestDtype_int16.test_dtypes _____________________________________________
```

--- a/pandas/tests/window/test_expanding.py
+++ b/pandas/tests/window/test_expanding.py
@@ -137,6 +137,14 @@ def test_expanding_count_default_min_periods_with_null_values(constructor):
     tm.assert_equal(result, expected)
 
 
+@pytest.mark.parametrize("constructor", [Series, DataFrame])
+def test_expanding_count_with_min_periods_exceeding_series_length(constructor):
+    # GH 25857
+    result = constructor(range(5)).expanding(min_periods=6).count()
+    expected = constructor([np.nan, np.nan, np.nan, np.nan, np.nan])
+    tm.assert_equal(result, expected)
+
+
 @pytest.mark.parametrize(
     "df,expected,min_periods",
     [

